addons/website/models/website_visitor.py

```diff
@@ -28,14 +28,30 @@ class WebsiteTrack(models.Model):
 class WebsiteVisitor(models.Model):
     _name = 'website.visitor'
     _description = 'Website Visitor'
-    _order = 'last_connection_datetime DESC'
+    _order = 'id DESC'
 
-    name = fields.Char('Name')
-    access_token = fields.Char(required=True, default=lambda x: uuid.uuid4().hex, copy=False, groups='website.group_website_publisher')
-    active = fields.Boolean('Active', default=True)
+    def _get_access_token(self):
+        """ Either the user's partner.id or a hash. """
+        assert request, _("Visitors can only be created through the frontend.")
+
+        if not self.env.user._is_public():
+            return self.env.user.partner_id.id
+
+        msg = repr((
+            request.httprequest.remote_addr,
+            request.httprequest.environ.get('HTTP_USER_AGENT'),
+            fields.Date.today().strftime('%Y-%m'),
+            self.env["ir.config_parameter"].sudo().get_param("database.secret"),
```

Comment:
Please don't use the db secret for that.
1) it's not used in a proper HMAC scheme, so it's not considered safe enough. SHA1 is known to be broken for security purposes. 
2) why do you even need a secret here? if you want to make sure visitor IDs are different on different websites, then this should be keyed on the website domain, not the db secret. And apparently you don't care about that when the partner ID is used? (also trivial to forge, btw?)